okdownload-connection-okhttp/src/main/java/com/liulishuo/okdownload/core/connection/DownloadOkHttp3Connection.java

```diff
@@ -103,6 +103,14 @@
         return response == null ? null : response.header(name);
     }
 
+    @Override
+    public String getRedirectLocation() {
```

Comment:
Why `request.url` always be redirect-location?

Reply:
There is the comment of `Response#request` : 
```
It may be the request generated in response to an HTTP redirect or authentication challenge. In this case the request URL may be different than the initial request URL.
```
Detail info is [here](https://square.github.io/okhttp/3.x/okhttp/okhttp3/Response.html#request--)

Reply:
`response.request()` returns final request of this connection task. If the original url connection is a redirect connection, then the result of `response.request()` is redirect location request instead of original url, and the latter can be got through `response.priorResponse().request()`. In a word, the there is redirect occurring, `response.request()` returns redirect location request,   otherwise it is original url's request. So this is what the "may" means in my opinion.

I agree with you that the method name will be confused here.

Reply:
I knew what is `response.request()` get, all My point is only the method name will be confused further, please change the method name or return the 'real' redirect location, if there isn't any redirect happen, please return `null`.

Reply:
I knew what is `response.request()` get, all My point is only the method name will be confused further, please change the method name or return the 'real' redirect location, if there isn't any redirect happen, please return `null`.

Reply:
I knew what is `response.request()` get, all My point is only the method name will be confused further, please change the method name or return the 'real' redirect location, if there isn't any redirect happen, please return `null`.

Reply:
I have changed implementation of `getRedirectLocation` so that this method is more make sense.

Reply:
I have changed implementation of `getRedirectLocation` so that this method is more make sense.

Reply:
I have changed implementation of `getRedirectLocation` so that this method is more make sense.

Reply:
I have changed implementation of `getRedirectLocation` so that this method is more make sense.

Reply:
I have changed implementation of `getRedirectLocation` so that this method is more make sense.

Reply:
I knew what is `response.request()` get, all My point is only the method name will be confused further, please change the method name or return the 'real' redirect location, if there isn't any redirect happen, please return `null`.

Reply:
I have changed implementation of `getRedirectLocation` so that this method is more make sense.